app/views/proposals/_tooltip.html.haml

```diff
@@ -1,11 +1,12 @@
 - progress_status = event.progress_status
 %ul.list-unstyled
-  %li{'class'=>class_for_todo(progress_status['registered'])}
-    %span{'class'=>icon_for_todo(progress_status['registered'])}
-      - if progress_status['registered']
-        Speaker(s) registered to the conference
-      - else
-        = link_to 'Speaker(s) not registered to the conference', new_conference_conference_registration_path(event.program.conference.short_title)
+  - if @conference.registration_open? || (@conference.user_registered? current_user)
```

Comment:
https://github.com/openSUSE/osem/blob/master/app/models/ability.rb#L69 is used to check if user can register or not, but it will also return true in some cases even when speaker have already registered which is wrong. The code according to me should be :
 `(conference.registration_open? && !conference.registration_limit_exceeded? || conference.program.speakers.confirmed.include?(user))&&!(conference.user_registered? user)`
Shouldn't it be?
And we cannot use CanCanCan in https://github.com/namangupta01/osem/blob/master/app/views/proposals/_tooltip.html.haml#L3 and https://github.com/namangupta01/osem/blob/master/app/views/proposals/index.html.haml#L77 because we are not checking if a user can register here or not here.
We can only use CanCanCan in https://github.com/namangupta01/osem/blob/master/app/views/proposals/index.html.haml#L51 because we are checking here if a user can make a registration or not.
Please, let me know if i am wrong?